### ISSUE TITLE
about PGD attack

### ISSUE THREAD

**Issue by yearep7** [bethgelab/foolbox#697]
As for the pgd attack, no matter how large or small the disturbance value I modify, the quality of its image generation seems to be the same. What is the wrong parameter setting that causes the disturbance change to have no effect

**Comment by zimmerrol** [comment]
Hi @yearep7, can you please post a minimal working example that reproduces your findings? Then we can look at your parameters (and how you use the attack) to check what's wrong.

**Comment by yearep7** [comment]
`fmodel = foolbox.models.PyTorchModel(net, bounds=(0, 1), num_classes=num_classes,device='cuda:0')
# FGSM
batch_start_time = time.time()

data=[]
for batch_idx, (inputs, target) in tqdm(enumerate(testloader)):
    data.append((inputs, target))

def save(index):
    print(index)
    inputs=data[index][0]
    target = data[index][1]
    image = inputs.detach().numpy()
    label = target.detach().numpy()
    # apply attack on source image
    attack = foolbox.attacks.PGD(fmodel)
    adversarial = attack(image, label)
    if adversarial is None:
        print('wu attack')
        adversarial=inputs

    for i in range(len(adversarial)):
        adversarial1=255*torch.from_numpy(adversarial[i].transpose(1,2,0))
        #
        name=class_name[label[i]]
        if not os.path.exists(model_folder+'/'+name):
            os.mkdir(model_folder+'/'+name)
        adversarial1=np.uint8(adversarial1)
        adversarial1=cv2.cvtColor(adversarial1,cv2.COLOR_RGB2BGR)
        cv2.imencode('.png', adversarial1)[1].tofile(model_folder+'/'+name+'/'+name+'_%d_%d.png'%(index,i))
            # cv2.imshow('test',adversarial1)
            # cv2.waitKey(1 )
#
No matter how I change the parameters, it operates according to a specified disturbance`

**Comment by yearep7** [comment]
`fmodel = foolbox.models.PyTorchModel(net, bounds=(0, 1), num_classes=num_classes,device='cuda:0')
# pgd
batch_start_time = time.time()

data=[]
for batch_idx, (inputs, target) in tqdm(enumerate(testloader)):
    data.append((inputs, target))

def save(index):
    print(index)
    inputs=data[index][0]
    target = data[index][1]
    image = inputs.detach().numpy()
    label = target.detach().numpy()
    # apply attack on source image
    attack = foolbox.attacks.PGD(fmodel)
    adversarial = attack(image, label)
    if adversarial is None:
        print('wu attack')
        adversarial=inputs

    for i in range(len(adversarial)):
        adversarial1=255*torch.from_numpy(adversarial[i].transpose(1,2,0))
        #
        name=class_name[label[i]]
        if not os.path.exists(model_folder+'/'+name):
            os.mkdir(model_folder+'/'+name)
        adversarial1=np.uint8(adversarial1)
        adversarial1=cv2.cvtColor(adversarial1,cv2.COLOR_RGB2BGR)
        cv2.imencode('.png', adversarial1)[1].tofile(model_folder+'/'+name+'/'+name+'_%d_%d.png'%(index,i))
            # cv2.imshow('test',adversarial1)
            # cv2.waitKey(1 )
#`
No matter how I change the parameters, it operates according to a specified disturbance
